Add overload getSoCFraction()

pmskintner opened the pull request:
- Returns value calculated from logic with the set temperatures.
- Remove unused node weight function

pmskintner committed:
Add overload getSoCFraction()

- Returns value calculated from logic with the set temperatures.
- Remove unused node weight function

chipbarnaby reviewed (commented):
Can't you save the most recently calculated SoC value in the class?  Then the getSoCFraction just returns that value.  Or am I missing something?

Change the primrary fcn name to CalcSocCFraction() perhaps.  Then getSoCFraction() is a simple getter.  Or make a new variant getLatestSoCFraction() that is the simple getter.

pmskintner committed:
Merge branch 'master' into getSoCFraction_overload

pmskintner commented:
> Can't you save the most recently calculated SoC value in the class? Then the getSoCFraction just returns that value. Or am I missing something?
> 
> Change the primrary fcn name to CalcSocCFraction() perhaps. Then getSoCFraction() is a simple getter. Or make a new variant getLatestSoCFraction() that is the simple getter.

@chipbarnaby, definitely possible to do. I'll just have to watch out for the case where a call to calcSoCFraction(double tMains_C, double tMinUseful_C), does not set the current SoC based off the inputs.

pmskintner committed:
Refactor to calc SoC only when needed

- Member variable currentSoCFraction with a getter.
- And updates when necessary not in every call to a SoC logic getTankValue

pmskintner commented:
@chipbarnaby  updated, and cut a bunch of calls calculating the SoC
